resources/event-sources/templates/100-pre-upgrade-job.yaml

```diff
@@ -0,0 +1,136 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: event-sources-pre-upgrade
+  namespace: kyma-system
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-weight: "0"
+---
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Since `helm del --purge` is used, is `kubectl delete` for each resource needed?

Reply:
The job can be run multiple times and it tries to delete the Event-Bus resources, also if only some of them partially exists. This could be the case if something happened during upgrade. From the perspective of this job, it could be started again cause it works on incomplete deleted Event-Bus resources too.

Reply:
What do you think can go wrong in `helm del --purge`? It is not clear to me why `kubectl delete` is needed here. It would make the job more concise and the Role/RoleBinding is not needed as well.

This job is meant to be executed once(`restartPolicy: Never`). https://github.com/kyma-project/kyma/pull/7787/files#diff-48334421faca654acd48a0ada21d19d0R102

Reply:
See my early comments. Only Helm is used now.